### PR TITLE
Feature/recent matches into master

### PR DESCRIPTION
**Pull request by sam-castaneda**

### CODE REVIEW

File: src/main/java/com/accesso/challengeladder/controller/MatchController.java [hunk 1]
```diff
@@ -35,6 +32,11 @@ public MatchController() throws SQLException, IOException
             return JsonUtil.toJson(matchService.getMatch(req.params(":id")));
         });
 
+        // get information of recent matches
+        get("/recent_matches/:limit", (req, res) -> {
```
Comment (olopez-accesso) [inline src/main/java/com/accesso/challengeladder/controller/MatchController.java]: the resource should be matches and not recent_matches. IMO frontend should paginate the existing response from GET /matches

File: src/main/java/com/accesso/challengeladder/model/User.java [hunk 2]
```diff
@@ -10,7 +10,7 @@
     @DatabaseField(generatedId = true)
     private int id;
 
-    @DatabaseField
+    @DatabaseField(columnName = "name")
```
Comment (olopez-accesso) [inline src/main/java/com/accesso/challengeladder/model/User.java]: no need to specify the columnName as it matches the attribute name

File: src/main/java/com/accesso/challengeladder/services/MatchService.java [hunk 3]
```diff
@@ -234,6 +234,22 @@ public Match updateMatch(String matchId, String victorId, Date matchTimestamp, S
         return matchList;
     }
 
+    public List<Match> getRecentMatches(int limit) throws SQLException
```
Comment (olopez-accesso) [inline src/main/java/com/accesso/challengeladder/services/MatchService.java]: I would remove this call/service function

**Comment by sam-castaneda** [comment]
@olopez-accesso I guess I should have put a brief description of what this is for.
We wanted to put a small section next to the Rankings to show the most recent matches, where the front-end can put a limit on how many they want to see.

It looks like you don't like the new GET /recent_matches ...
What is your suggestion then to change the existing url, because GET /matches/1 would return match with id = 1.

Should we do GET /matches/limit/:limit ?

**Comment by olopez-accesso** [comment]
why not retrieve all from BE and just apply the limit on the FE?

we could use 
http://192.168.2.46:4567/matches
and order by match_date DESC in the BE if we don't want the FE to do the ordering.

**Comment by sam-castaneda** [comment]
Because that call will grow with every match, it's not necessary to make these huge calls when you can easily define how many you want to display anyway.
Do we really need to pull 100 matches, if we only need to display 10-20?
Or in a year from now, do we really need to pull 1000 matches?

**Comment by sam-castaneda** [comment]
Adding a body to a GET request is possible, but generally something that should be avoided:
http://stackoverflow.com/questions/978061/http-get-with-request-body

Adding a /limit/:limit param is not that uncommon.

**Comment by sam-castaneda** [comment]
We could also just add pagination
/matches?page=x
where /matches would default to page=1, and then set a limit of 20 or 100 or whatever as the limit per page.

**Comment by olopez-accesso** [comment]
I agree with no adding body content to the GET request. 
I'll add the limit as in-url param as well like: GET /matches?page=1&limit=10

**Comment by jfonte-accesso** [comment]
Since I do not like doing unnecessary math, page 0 is the first page (or don't pass it). That way the query offset is page*limit instead of (page-1) * limit only if page > 0.

**Comment by jfonte-accesso** [comment]
Changes should be taken care of.
@olopez-accesso can you confirm?

**Pull request merged**